modules/apps/frontend-js/frontend-js-spa-web/src/main/resources/META-INF/resources/senna/app/App.js

```diff
@@ -1360,7 +1356,13 @@ class App extends EventEmitter {
 	 * @return {boolean} True if an element was removed.
 	 */
 	removeRoute(route) {
-		return array.remove(this.routes, route);
+		const routeIndex = this.routes.indexOf(route);
+
+		if (routeIndex >= 0) {
+			Array.prototype.splice.call(this.routes, routeIndex, 1);
```

Comment:
`this.routes` seems guaranteed to be an array, so why are we doing the `call` trick here? Could this not just be `this.routes.splice(routeIndex, 1)`?

Reply:
Probably, yeah... just wanted to be safe as this already introduces enough uncertainty :(

Reply:
https://github.com/liferay-frontend/liferay-portal/pull/489/commits/931de2ae8dee76321bf1d09ca6b3aea092322272